src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,52 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    public static int splitAndSum(String text) {
+        if (isNullOrBlank(text)) {
+            return 0;
+        }
+
+        int[] numbers = toInts(toTokens(text));
+        checkNegative(numbers);
+        return sum(numbers);
+    }
+
+    private static boolean isNullOrBlank(String text) {
+        return text == null || text.isBlank();
+    }
+
+    private static String[] toTokens(String text) {
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
```

Comment:
컴파일할 패턴이 어떠한 역할을 하는 패턴인지 상수로 추출하면서 이름지어보면 어떨까요? 코드를 읽는 입장에서 더 이해하기 좋은 코드가 될 수 있을 것 같아요!!

Reply:
추가로 단순히 내부에서 사용되는 변수라도 명확한 이름을 적어주세요!!

```suggestion
        Matcher matcher = Pattern.compile("//(.)\n(.*)").matcher(text);
```